### PR TITLE
Use IP failover for router

### PR DESCRIPTION
**Pull request by lebauce**

### CODE REVIEW

File: templates/var/lib/ansible/playbooks/main.yml [hunk 1]
```diff
@@ -38,6 +38,8 @@
   - name: Restart node service
     service: name={{openshift.common.service_type}}-node state=restarted
 
+- include: /var/lib/ansible/playbooks/ipfailover.yml
+
```
Comment (jprovaznik) [inline templates/var/lib/ansible/playbooks/main.yml:44]: Would it make sense (if possible) to make ipfailover setup only when neutron lbaas is used? IOW move router_vip resources into loadbalancer_neutron.yaml and run this playbook only when loadbalancer_type==neutron?

File: master.yaml [hunk 2]
```diff
@@ -229,6 +234,10 @@ resources:
       security_groups:
       - {get_param: master_security_group}
       network: {get_param: fixed_network}
+      allowed_address_pairs:
+      - ip_address: "224.0.0.18"
+        mac_address: "01:00:5e:00:00:12"
+      - ip_address: {get_param: router_vip}
```
Comment (jprovaznik) [inline master.yaml]: could you please add a comment about this fake ip?

File: templates/var/lib/ansible/playbooks/ipfailover.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+{{=<% %>=}}
+- hosts: masters[0]
+  sudo: yes
+  tasks:
+  - name: Deploy Openshift IP failover for router
+    command: oadm ipfailover --create --service-account=router --interface=eth0 --selector='{{ openshift_router_selector }}' --replicas={{ num_infra }} --virtual-ips="{{ router_vip }}" --credentials=/etc/origin/master/openshift-router.kubeconfig
```
Comment (jprovaznik) [inline templates/var/lib/ansible/playbooks/ipfailover.yml:7]: openshift-ansible is failing on this for me:
GATHERING FACTS **************************************************************\* 
<ipf-openshift-master-0.example.com> ESTABLISH CONNECTION FOR USER: cloud-user
<ipf-openshift-master-0.example.com> REMOTE_MODULE setup
<ipf-openshift-master-0.example.com> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=cloud-user -o ConnectTimeout=10 ipf-openshift-master-0.example.com /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1466506793.16-199816308477047 && echo $HOME/.ansible/tmp/ansible-tmp-1466506793.16-199816308477047'
<ipf-openshift-master-0.example.com> PUT /tmp/tmpywy_m5 TO /home/cloud-user/.ansible/tmp/ansible-tmp-1466506793.16-199816308477047/setup
<ipf-openshift-master-0.example.com> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=cloud-user -o ConnectTimeout=10 ipf-openshift-master-0.example.com /bin/sh -c 'sudo -k && sudo -H -S -p "[sudo via ansible, key=lybvoommngfsitjrmvukiuauaoknxkbs] password: " -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-lybvoommngfsitjrmvukiuauaoknxkbs; LANG=C LC_CTYPE=C /usr/bin/python /home/cloud-user/.ansible/tmp/ansible-tmp-1466506793.16-199816308477047/setup; rm -rf /home/cloud-user/.ansible/tmp/ansible-tmp-1466506793.16-199816308477047/ >/dev/null 2>&1'"'"''
ok: [ipf-openshift-master-0.example.com]

TASK: [Deploy Openshift IP failover for router] ******************************\* 
<ipf-openshift-master-0.example.com> ESTABLISH CONNECTION FOR USER: cloud-user
fatal: [ipf-openshift-master-0.example.com] => One or more undefined variables: 'openshift_router_selector' is undefined

FATAL: all hosts have already failed -- aborting

PLAY RECAP *******************************************************************\* 
           to retry, use: --limit @/root/main.retry

ipf-openshift-master-0.example.com : ok=515  changed=122  unreachable=1    failed=0  
ipf-openshift-master-1.example.com : ok=308  changed=72   unreachable=0    failed=0  
ipf-openshift-master-2.example.com : ok=308  changed=72   unreachable=0    failed=0  
ipf-openshift-node-m1y0p193.example.com : ok=129  changed=28   unreachable=0    failed=0  
localhost                  : ok=46   changed=6    unreachable=0    failed=0

File: openshift.yaml [hunk 4]
```diff
@@ -335,6 +335,19 @@ resources:
...
+    properties:
+      network_id: {get_resource: fixed_network}
+      fixed_ips:
+       - subnet: {get_resource: fixed_subnet}
+
+  router_floating_ip:
+    type: OS::Neutron::FloatingIP
+    properties:
+      floating_network: {get_param: external_network}
+      port_id: {get_resource: router_vip_port}
```
Comment (jprovaznik) [inline openshift.yaml]: we should probably show this value in stack output so user can easily find it

File: env_origin.yaml [hunk 5]
```diff
@@ -21,3 +21,4 @@ parameters:
 resource_registry:
   OOShift::LoadBalancer: loadbalancer_neutron.yaml
   OOShift::ContainerPort: sdn_openshift_sdn.yaml
+  OOShift::IPFailover: ipfailover_noop.yaml
```
Comment (jprovaznik) [inline env_origin.yaml]: should be "ipfailover_none.yaml"

Comment (lebauce) [inline env_origin.yaml]: Oops. Thanks

File: ipfailover_keepalived.yaml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    type: OS::Neutron::FloatingIP
+    properties:
+      floating_network: {get_param: external_network}
+      port_id: {get_resource: router_vip_port}
+
+outputs:
+  router_vip:
+    description: >
+      The Openshift Router Virtual IP
+    value: {get_attr: [router_vip_port, fixed_ips, 0, ip_address]}
```
Comment (jprovaznik) [inline ipfailover_keepalived.yaml:51]: an empty string is returned, this change worked for me:

```
-    value: {get_attr: [router_vip_port, fixed_ips, 0, ip_address]}
+    value: {get_attr: [router_floating_ip, floating_ip_address]}
```

**Pull request merged**